### DISCUSSION TITLE
Provide an easy way to generalize Github actions across organization repositories

### DISCUSSION THREAD

**Discussion by benoittgt** [community/community#85004] [Actions]
### Select Topic Area

Product Feedback

### Body

Hello

I am working for a company and we have around +300 repositories. Recently we wanted to introduce [actions/stale](https://github.com/actions/stale) but on all our repositories. Github does not provide an easy way to do it.

It was mentioned in [Github Universe 2022](https://www.youtube.com/watch?v=2Tv-qm_hdCw) but it requires IaC and Pulumi.

Before actions/stale it was a probot. The "cleanest" way was to create a file in `https://github.com/my-org/.github/.github/reuse.yml`, then from other repositories, use the [_extends](https://probot.github.io/docs/best-practices/#store-configuration-in-the-repository) attributes, to import the yml. 
Now with the new `actions/stale` you have a to do more work, and use `uses:` [[doc](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsuses)].

Github should leverage `.github` root repo where you could define an action that will be run on each repositories.

There is other proposal at the moment:
- From one repository, use a Github App token to loop through repositories and run the action https://github.com/actions/stale/discussions/1012#discussioncomment-6393131
  - https://github.com/actions/stale/discussions/1089#discussioncomment-7239931

**Comment by ReenigneArcher** [comment]
GitHub had a beta feature "required workflows", which allowed easily reusing workflows across repos. Sadly it only worked for pull requests and no other triggers from my recollection... and they also discontinued it. https://docs.github.com/en/enterprise-server@3.11/actions/sharing-automations/required-workflows

With 300+ repos, it would probably be better to run `actions/stale` in a single workflow instead of 300+ times simultaneously. At least if you care about runner consumption/usage. This is what we're doing for our org. https://github.com/LizardByte/.github/blob/master/.github/workflows/issues-stale.yml

It first collects the repos in the org, then runs a strategy matrix to check all the stale issues and PRs. To me this has a few advantages over copying the workflow to every repo.

1. Only consumes 1 runner (of our free allotment of 20)
2. Less updates to the workflow in every repo.

If you don't want to use the fork/pr branch of actions/stale that allows passing in a different user and repo, then you could replicate your workflow(s) to your repos. I use this to replicate some common workflows, like linters and whatnot. https://github.com/derberg/manage-files-in-multiple-repositories

I agree your feature request is a nice idea. I doubt GitHub will ever see this though, so the above are just workarounds for you to consider.
